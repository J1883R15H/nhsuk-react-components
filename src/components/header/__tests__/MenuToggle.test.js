import React from 'react';
import { mount } from 'enzyme';
import MenuToggle from '../components/menutoggle';
import HeaderContext from '../HeaderContext';

const testContext = {
  toggleMenu: jest.fn()
};

describe('Header.MenuToggle', () => {
  it('matches snapshot', () => {
    const wrapper = mount(
      <HeaderContext.Provider value={testContext}>
        <MenuToggle></MenuToggle>
      </HeaderContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
  it('calls toggleMenu on click', () => {
    const wrapper = mount(
      <HeaderContext.Provider value={testContext}>
        <MenuToggle />
      </HeaderContext.Provider>
    );
    wrapper.find('#toggle-menu').simulate('click');
    expect(testContext.toggleMenu).toHaveBeenCalled();
    wrapper.unmount();
  });
});
