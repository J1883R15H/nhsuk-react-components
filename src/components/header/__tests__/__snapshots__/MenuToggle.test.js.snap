// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.MenuToggle matches snapshot 1`] = `
<MenuToggle
  className=""
>
  <div
    className="nhsuk-header__menu"
  >
    <button
      aria-controls="header-navigation"
      aria-label="Open menu"
      className="nhsuk-header__menu-toggle"
      id="toggle-menu"
      onClick={[MockFunction]}
      type="button"
    >
      Menu
    </button>
  </div>
</MenuToggle>
`;
